src/gql/schema.rs

```diff
@@ -0,0 +1,626 @@
+					let fd_name = fd_name.clone();
+					FieldFuture::new(async move {
+						let record = ctx.parent_value.as_value().unwrap();
+						let GqlValue::Object(record_map) = record else {
+							todo!("got unexpected: {record:?}, processing field {fd_name}")
+						};
+						let val = record_map.get(&fd_name).unwrap();
+
+						Ok(Some(val.to_owned()))
+					})
+				}));
+		}
+
+		types.push(Type::Object(table_ty_obj));
+		types.push(table_order.into());
+		types.push(Type::Enum(table_orderable));
+		trace!("\n\n\n{table_filter:?}");
+		types.push(Type::InputObject(table_filter));
+	}
+
+	trace!("current Query object for schema: {:?}", query);
+
+	let mut schema = Schema::build("Query", None, None).register(query);
+	for ty in types {
+		trace!("adding type: {ty:?}");
+		schema = schema.register(ty);
+	}
+
+	let id_interface =
+		Interface::new("record").field(InterfaceField::new("id", TypeRef::named_nn(TypeRef::ID)));
+	schema = schema.register(id_interface);
+
+	// TODO: when used get: `Result::unwrap()` on an `Err` value: SchemaError("Field \"like.in\" is not sub-type of \"relation.in\"")
+	let relation_interface = Interface::new("relation")
+		.field(InterfaceField::new("id", TypeRef::named_nn(TypeRef::ID)))
+		.field(InterfaceField::new("in", TypeRef::named_nn("record")))
+		.field(InterfaceField::new("out", TypeRef::named_nn("record")))
+		.implement("record");
+	schema = schema.register(relation_interface);
+
+	let res = schema.finish();
+	match res {
+		Ok(s) => Ok(s),
+		Err(_) => {
+			warn!("note: Query reloading is not yet implemented. This means that the schema is generated on startup, so if starting in memory the schema will always be empty");
+			res.unwrap();
+			unreachable!();
+		}
+	}
+
+	// Ok(schema.finish().unwrap())
+}
+
+fn sql_value_to_gql_value(v: SqlValue) -> Result<GqlValue, ()> {
+	let out = match v {
+		SqlValue::None => GqlValue::Null,
+		SqlValue::Null => GqlValue::Null,
+		SqlValue::Bool(b) => GqlValue::Boolean(b),
+		SqlValue::Number(n) => match n {
+			surrealdb::sql::Number::Int(i) => GqlValue::Number(i.into()),
+			surrealdb::sql::Number::Float(f) => GqlValue::Number(Number::from_f64(f).ok_or(())?),
+			surrealdb::sql::Number::Decimal(_) => todo!("surrealdb::sql::Number::Decimal(_)"),
+			_ => todo!(),
+		},
+		SqlValue::Strand(s) => GqlValue::String(s.0),
+		SqlValue::Duration(d) => GqlValue::String(d.to_string()),
+		SqlValue::Datetime(d) => GqlValue::String(d.to_string()),
+		SqlValue::Uuid(uuid) => GqlValue::String(uuid.to_string()),
+		SqlValue::Array(a) => {
+			GqlValue::List(a.into_iter().map(|v| sql_value_to_gql_value(v).unwrap()).collect())
+		}
+		SqlValue::Object(o) => GqlValue::Object(
+			o.0.into_iter()
+				.map(|(k, v)| (Name::new(k), sql_value_to_gql_value(v).unwrap()))
+				.collect(),
+		),
+		SqlValue::Geometry(_) => todo!("SqlValue::Geometry(_) "),
+		SqlValue::Bytes(_) => todo!("SqlValue::Bytes(_) "),
+		SqlValue::Thing(t) => GqlValue::String(t.to_string()),
```

Comment:
Do we plan on introducing a custom type for record id's?

Reply:
I'm using a combination of GraphQL's ID type and specific table types